lib/guardian_extensions.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+module GuardianExtensions
+  def can_see_topic?(topic, hide_deleted = true)
+    if SiteSetting.encrypt_enabled? && topic.private_message? && topic.is_encrypted?
```

Comment:
```suggestion
    if SiteSetting.encrypt_enabled? && topic&.is_encrypted?
```

I think you can simplify this as `is_encrypted?` already checks if `private_message?`.

Also, an admin could disable `discourse-encrypt` and bypass this check anyway. Is it for security purposes or did you want to short-circuit evaluation?

I believe `topic` can be `nil` too which might cause some problems. I see we handle this case in core:

```ruby
  def can_see_topic?(topic, hide_deleted = true)
    return false unless topic
    # ...
  end
```

Reply:
Rather than overwriting the method, I would recommend aliasing it. That way discourse-encrypt can call it, and, if true, check if encrypted as well. It's nicer than a freedom patch because if the underlying method changes you continue to get those updates.